LanguageFeatures/Augmentation-libraries/augmenting_types_A01_t03_lib.dart

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion It is a compile-time error if:
+///
+/// - The augmenting type and corresponding type are not the same kind: class,
+///   mixin, enum, or extension. You can't augment a class with a mixin, etc.
+///
+/// @description Checks that it is not an error if an augmenting type and the
+/// corresponding type are the same kind. Test type aliases
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
+
+augment library 'augmenting_types_A01_t03.dart';
+
+augment class CAlias {
+  String foo() => "C";
+}
+
+augment mixin MAlias {
+  String foo() => "M";
+}
+
+augment enum EAlias {
+  e1;
```

Comment:
This is going to be interesting! ;-)

I would expect that this is an error because there is an `e1` in `E` already (so it would fail when we try to add another declaration named `e1`).

However, the feature specification does use a construct which would be `augment e1` in this case, which would then be implied to be OK.

Reply:
Changed to `augment e1;`